### ISSUE TITLE
how does validation split arrange the data in detail regarding mini-batch training?

### ISSUE THREAD

**Issue by pswpswpsw** [fchollet/keras#4692]
I have read the document but still have some doubt. 

**Data**
- For example, I have a 1024 total data. I don't have test data but only 1024 data: **validation_spit = 0.5**

**Optimization scheme**
- batch_size = 32, epoch is 100.
- shuffle=True, so that only training data will be shuffled. 

For each epoch, I will run 512/32=16 batch, with each batch_size = 32, that is to say, I will have total 16 weight updates for each epoch.

So when I say verbose = 1, I will see output of error of training and validation for each epoch. 

Here comes my question, what is exactly the validation data being used for each epoch? Is it never change for each epoch, as it was determined at the beginning of training process, and keep in a secret place only for "testing" the model since then? (remember we have a validation_split = 0.5) So I guess the validation data is the last some percentage of the raw data input?(training + validation data)? 

If that's the case, then we should **shuffle on raw data input** **before** the **shuffle** in model fit is necessary. 

Thanks!
Shawn

**Comment by fchollet** [comment]
This is explained in the docs.

> what is exactly the validation data being used for each epoch? Is it never change for each epoch, as it was determined at the beginning of training process, and keep in a secret place only for "testing" the model since then? (remember we have a validation_split = 0.5) So I guess the validation data is the last some percentage of the raw data input?(training + validation data)?

Yes. The validation split is the second half of the data samples you passed to the call to `fit`.

> If that's the case, then we should shuffle on raw data input before the shuffle in model fit is necessary.

No shuffle happens before cutting the validation split. Again, this is in the docs. Take care of shuffling the data yourself if necessary.
